collector/collector.go

```diff
@@ -3,161 +3,109 @@ package collector
 import (
 	"context"
 	"encoding/json"
+	"errors"
 	"fmt"
 	"net/http"
-	"strconv"
 
+	"github.com/go-openapi/runtime"
+	"github.com/lidofinance/terra-monitors/client"
 	"github.com/sirupsen/logrus"
 )
 
 type Metrics string
 
-var (
-	BlunaBondedAmount Metrics = "bluna_bonded_amount"
-	BlunaTotalSupply  Metrics = "bluna_total_supply"
-	GlobalIndex       Metrics = "global_index"
-)
-
-func parseRequestBody(resp *http.Response, ret interface{}) error {
-	m := struct {
-		Result interface{} `json:"result"`
-		Error  string      `json:"error"`
-	}{
-		Result: ret,
+func CastMapToStruct(m interface{}, ret interface{}) error {
+	data, err := json.Marshal(m)
+	if err != nil {
+		return fmt.Errorf("failed to marshal body interface{}: %w", err)
 	}
-	decoder := json.NewDecoder(resp.Body)
-	err := decoder.Decode(&m)
+	err = json.Unmarshal(data, ret)
 	if err != nil {
 		return fmt.Errorf("failed to unmarshal response body: %w", err)
 	}
-	if m.Error != "" {
-		return fmt.Errorf(m.Error)
-	}
 	return nil
 }
 
 type Collector interface {
 	Get(metric Metrics) (float64, error)
```

Comment:
I think the Metrics **type** should be renamed to Metric

Reply:
done